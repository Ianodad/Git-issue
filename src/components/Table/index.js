import React from 'react'
import { NavLink } from 'react-router-dom';

import {  BiLinkAlt } from "react-icons/bi";
var dayjs = require('dayjs')


const Index=({style, data, repo}) => {
   
  
  // console.log(data)
  return (
      <div className="container-fluid">
        <table className="table">
        <thead className="" style={{backgroundColor: '#F1F0FF'}}>
          { repo ? (<tr>
            <th scope="col">Name</th>
            <th scope="col">Has Issues</th>
            <th scope="col">Date</th>
            <th scope="col">Open Issue Count</th>
            <th scope="col">Link</th>
            <th scope="col">Detail</th>
          </tr>): (
            <tr>
            <th scope="col">No</th>
            <th scope="col">Title</th>
            <th scope="col">State</th>
            <th scope="col">User</th>
            <th scope="col">Link</th>
            <th scope="col">Detail</th>
          </tr>
          )}
        </thead>
        <tbody>
          {data && data.map((item, key)=>{
            if (repo) return (
            <tr key={key}>
            <td scope="row">{item.name}</td>
            <td>{item.has_issues}</td>
            <td>{dayjs(item.created_at).format('DD/MM/YYYY')}</td>
            <td>{item.open_issues_count}</td>
            <td><a href={item.html_url} target="_blank" className="nav-item nav-link"><BiLinkAlt/></a></td>
            <td><NavLink to={`/issues/${item.owner.login}/${item.name}`} type="button" class="btn" style={{backgroundColor:"#443266", color:"white"}}>More</NavLink></td>
          </tr>
            )
            return (
          <tr key={key}>
            <td scope="row">{item.number}</td>
            <td>{item.title}</td>
            <td>{item.state}</td>
            <td>{item.user.login}</td>
            <td><a href={item.html_url} target="_blank" className="nav-item nav-link"><BiLinkAlt/></a></td>
            <td><NavLink to={`/inspection/${item.number}`} type="button" class="btn" style={{backgroundColor:"#443266", color:"white"}} >More</NavLink></td>

          </tr>
            )
          })}
        </tbody>
      </table>

      </div>
    );
}

export default Index
