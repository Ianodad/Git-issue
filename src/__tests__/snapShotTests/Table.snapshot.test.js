import Table from "../../components/Table";
import renderer from 'react-test-renderer';

import { BrowserRouter, Switch } from 'react-router-dom';


describe("<Table/> testing component", () => {
    let component
    
    beforeEach(() => {
    component = renderer.create(
    <BrowserRouter>
        <Switch>
        <Table/>
        </Switch>
    </BrowserRouter>
    );
  });

    describe("Check component renders", () => {
         it('should render snapshot', () => {
          expect(component.toJSON()).toMatchSnapshot();
        });
        
    });
});