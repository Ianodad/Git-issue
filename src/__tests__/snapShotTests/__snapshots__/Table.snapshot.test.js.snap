// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table/> testing component Check component renders should render snapshot 1`] = `
<div
  className="container-fluid"
>
  <table
    className="table"
  >
    <thead
      className=""
      style={
        Object {
          "backgroundColor": "#F1F0FF",
        }
      }
    >
      <tr>
        <th
          scope="col"
        >
          No
        </th>
        <th
          scope="col"
        >
          Title
        </th>
        <th
          scope="col"
        >
          State
        </th>
        <th
          scope="col"
        >
          User
        </th>
        <th
          scope="col"
        >
          Link
        </th>
        <th
          scope="col"
        >
          Detail
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</div>
`;
